genie-web/src/main/java/com/netflix/genie/web/aspect/HealthCheckMetricsAspect.java

```diff
@@ -0,0 +1,248 @@
+            + "  void org.springframework.boot.actuate.health.AbstractHealthIndicator.doHealthCheck("
+            + "      org.springframework.boot.actuate.health.Health.Builder"
+            + "  )"
+            + ")"
+    )
+    @SuppressWarnings("checkstyle:IllegalThrows") // For propagating Throwable from joinPoint.proceed()
+    public void abstractHealthIndicatorDoHealthCheckMonitor(final ProceedingJoinPoint joinPoint) throws Throwable {
+        final long start = System.nanoTime();
+        Throwable throwable = null;
+        try {
+            joinPoint.proceed(joinPoint.getArgs());
+        } catch (final Throwable t) {
+            throwable = t;
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            recordHealthIndicatorTurnaround(turnaround, joinPoint, throwable);
+        }
+    }
+
+    private void recordHealthIndicatorTurnaround(
+        final long turnaround,
+        final ProceedingJoinPoint joinPoint,
+        @Nullable final Throwable throwable
+    ) {
+        log.debug(
+            "{} completed in {} ns (exception: {})",
+            joinPoint.getTarget().getClass().getSimpleName(),
+            turnaround,
+            throwable != null ? throwable.getClass().getSimpleName() : "none"
+        );
+        final Map<String, String> tags;
+        if (throwable == null) {
+             tags = MetricsUtils.newSuccessTagsMap();
+        } else {
+            tags = MetricsUtils.newFailureTagsMapForException(throwable);
+        }
+        tags.put(HEALTH_INDICATOR_CLASS_TAG_NAME, joinPoint.getTarget().getClass().getSimpleName());
+        registry
+            .timer(healthIndicatorTimerId.withTags(tags))
+            .record(turnaround, TimeUnit.NANOSECONDS);
+    }
+
+    /**
+     * Intercept calls to the main AbstractHealthAggregator and publish counters for number of invocation and failures,
+     * tagged with status and indicator name.
+     * @param joinPoint joinPoint for the aggregateDetails() call
+     */
+    @Before(
+        "execution("
+            + "  java.util.Map<String, Object> "
+            + "  org.springframework.boot.actuate.health.AbstractHealthAggregator.aggregateDetails("
+            + "    java.util.Map<String, org.springframework.boot.actuate.health.Health>"
+            + "  )"
+            + ")"
+    )
+    public void abstractHealthAggregatorAggregateDetailsMonitor(final JoinPoint joinPoint) {
+
+        final Map<String, Health> healthDetailsMap;
+        try {
+            @SuppressWarnings("unchecked")
+            final Map<String, Health> map = (Map<String, Health>) joinPoint.getArgs()[0];
+            healthDetailsMap = map;
+        } catch (final Throwable t) {
+            log.warn(
+                "Failed to cast AbstractHealthAggregator health details argument: {}",
+                joinPoint.getArgs(),
+                t
+            );
+            return;
+        }
+
+        healthDetailsMap.forEach(
```

Comment:
Don't we already have this via the lower level intercepts you have? (I'm forgetting how we discussed this before I left) I remember having a reason to do this at the global level but not sure why you're also doing it at the lower level. Seems like the top level one would be enough?

Reply:
Not sure what you mean by top level vs. lower level

Reply:
I believe we discussed this in person but I meant the aggregator (top) vs. the individual (lower) levels. I think we're good now